lib/pacemaker/pcmk_sched_notif.c

```diff
@@ -530,100 +530,108 @@ new_notify_entry(pe_resource_t *rsc, pe_node_t *node)
     return entry;
 }
 
-void
-collect_notification_data(pe_resource_t * rsc, gboolean state, gboolean activity,
-                          notify_data_t * n_data)
+/*!
+ * \internal
+ * \brief Add notification data for resource state and optionally actions
+ *
+ * \param[in] rsc        Clone or clone instance being notified
+ * \param[in] activity   Whether to add notification entries for actions
+ * \param[in] n_data     Notification data for clone
+ */
+static void
+collect_resource_data(pe_resource_t *rsc, bool activity, notify_data_t *n_data)
 {
+    GList *iter = NULL;
+    notify_entry_t *entry = NULL;
     pe_node_t *node = NULL;
 
-    if(n_data->allowed_nodes == NULL) {
+    if (n_data->allowed_nodes == NULL) {
         n_data->allowed_nodes = rsc->allowed_nodes;
     }
 
-    if (rsc->children) {
-        GList *gIter = rsc->children;
-
-        for (; gIter != NULL; gIter = gIter->next) {
-            pe_resource_t *child = (pe_resource_t *) gIter->data;
+    // If this is a clone, call recursively for each instance
+    if (rsc->children != NULL) {
+        for (iter = rsc->children; iter != NULL; iter = iter->next) {
```

Comment:
You don't need the if surrounding the for here.  If rsc->children is NULL, the loop will not run.